### PR TITLE
test: add unit tests for getSchemaAppEnsuringOrganization

### PR DESCRIPTION
**Pull request by rossiam**
* Added unit test for new `getSchemaAppEnsuringOrganization` function.
* Removed unnecessary generic specification for `APICommand` (which already default to `APICommandFlags`).

### CODE REVIEW

File: src/__tests__/lib/command/util/schema-util.test.ts [hunk 1]
```diff
@@ -11,9 +11,10 @@ import type {
 } from '@smartthings/core-sdk'
 
 import type { Profile } from '../../../../lib/cli-config.js'
+import type { stdinIsTTY, stdoutIsTTY } from '../../../../lib/io-util.js'
 import type { clipToMaximum, fatalError } from '../../../../lib/util.js'
 import type { APICommand } from '../../../../lib/command/api-command.js'
-import type { chooseOrganization, organizationDef } from '../../../../lib/command/util/organizations-util.js'
+import { chooseOrganization, organizationDef } from '../../../../lib/command/util/organizations-util.js'
```
Comment (Sitlintac) [inline src/__tests__/lib/command/util/schema-util.test.ts:17]: Here and on line 14 (`import type { stdinIsTTY, stdoutIsTTY }...`) the import statement isn't what I expected with regard to `type` and not `type`

File: src/__tests__/lib/command/util/schema-util.test.ts [hunk 2]
```diff
@@ -358,3 +395,106 @@ test('chooseSchemaAppFn uses correct endpoint to list schema apps', async () =>
...
+		expect(apiSchemaListMock).toHaveBeenCalledExactlyOnceWith()
+		expect(apiSchemaGetMock).toHaveBeenCalledExactlyOnceWith('not-in-list')
+
+		expect(fatalErrorMock).not.toHaveBeenCalled()
+		expect(chooseOrganizationMock).not.toHaveBeenCalled()
+		expect(apiOrganizationsGetMock).not.toHaveBeenCalled()
+		expect(cloneMock).not.toHaveBeenCalled()
+	})
+
+	it('sets and organization when app found without an organization', async () => {
```
Comment (Sitlintac) [inline src/__tests__/lib/command/util/schema-util.test.ts:438]: Little typo: "an organization"

File: src/__tests__/lib/command/util/schema-util.test.ts [hunk 3]
```diff
@@ -358,3 +395,106 @@ test('chooseSchemaAppFn uses correct endpoint to list schema apps', async () =>
...
+		chooseOrganizationMock.mockResolvedValueOnce('organization-id-1')
+		apiOrganizationsGetMock.mockResolvedValueOnce(organization1)
+		// TODO: maybe use an "updated organization" here and in corresponding expect below
```
Comment (Sitlintac) [inline src/__tests__/lib/command/util/schema-util.test.ts:441]: Just checking: this `TODO` is really for future work and not just an idea that you considered and rejected, right?

File: src/__tests__/lib/command/util/schema-util.test.ts (hunk 1 above)
Comment (rossiam) [reply src/__tests__/lib/command/util/schema-util.test.ts]: Yeah, I missed `type` on a bunch of things here. Fixed now.

File: src/__tests__/lib/command/util/schema-util.test.ts (hunk 2 above)
Comment (rossiam) [reply src/__tests__/lib/command/util/schema-util.test.ts]: Good catch. Fixed.

File: src/__tests__/lib/command/util/schema-util.test.ts (hunk 3 above)
Comment (rossiam) [reply src/__tests__/lib/command/util/schema-util.test.ts]: Didn't mean to leave that in. Removed now.

**Pull request merged**